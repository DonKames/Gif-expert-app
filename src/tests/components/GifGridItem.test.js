import { shallow } from 'enzyme';

import { GifGridItem } from "../../components/GifGridItem";


describe('Pruebas GifGridItem', () => {
    const title = 'Titulo';
    const url = 'https://media1.giphy.com/media/l0HlW7qQ7zqQkQq0c/giphy.gif';
    let wrapper = shallow(<GifGridItem title={title} url={url}/>);

    test('Debe mostrar el componente correctamente', () => {

        expect(wrapper).toMatchSnapshot();

    });

    test('Debe tener un <p></p> con el title', () => {
        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe( title );
    });
    
    test('Debe los datos de la img igual al de las props', () => {
        const img = wrapper.find('img');
        expect( img.prop( 'src' ) ).toBe( url );
        expect( img.prop( 'alt' ) ).toBe( title );
    });

    test('Debe tener animate__backInUp', () => {
        const div = wrapper.find('div');
        //expect( div.prop('className').indexOf('animate__backInUp') ).toBeGreaterThanOrEqual(0); el que hice yo
        //expect( div.prop( 'className' ).includes( 'animate__backInUp' ) ).toBe( true ); El que hizo el profe
        expect( div.hasClass( 'animate__backInUp' ) ).toBe( true );
        
    });
    
})
