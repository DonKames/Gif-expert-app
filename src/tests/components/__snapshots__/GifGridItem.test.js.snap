// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas GifGridItem Debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__backInUp"
>
  <img
    alt="Titulo"
    src="https://media1.giphy.com/media/l0HlW7qQ7zqQkQq0c/giphy.gif"
  />
  <p>
    Titulo
  </p>
</div>
`;
